pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -216,6 +227,33 @@ func (s *NodeController) SyncNodeStatus() error {
 	return nil
 }
 
+// CheckNodeReachable verifies if nodes are reachable by checking the time of last probe
+// and deletes pods from not reachable nodes.
+func (s *NodeController) CheckNodeReachable() error {
+	nodes, err := s.kubeClient.Nodes().List()
+	if err != nil {
+		return err
+	}
+	for _, node := range nodes.Items {
+		if util.Now().After(latestReadyTime(&node).Add(s.podEvictionTimeout)) {
+			s.deletePods(node.Name)
+		}
+	}
+	return nil
+}
+
+func latestReadyTime(node *api.Node) util.Time {
+	time := node.ObjectMeta.CreationTimestamp
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == api.NodeReady &&
+			condition.Status == api.ConditionFull &&
+			condition.LastProbeTime.After(time.Time) {
```

Comment:
Oh, I see.  You assign CreationTimestamp to time.... can we s/time/creationTime/?  overriding package name sounds weired.